integrations/gtag/lib/index.js

```diff
@@ -0,0 +1,622 @@
+      Integration.orderRefunded = Integration.orderRefundedEnhanced;
+      Integration.orderCompleted = Integration.orderCompletedEnhanced;
+      Integration.checkoutStepViewed = Integration.checkoutStepViewedEnhanced;
+      Integration.checkoutStepCompleted =
+        Integration.checkoutStepCompletedEnhanced;
+      Integration.orderUpdated = Integration.orderUpdatedEnhanced;
+
+      // Integration.productListFiltered = Integration.productListFilteredEnhanced;
+    }
+    /* eslint-enable */
+});
+
+/**
+ * Initialize.
+ *
+ * https://developers.google.com/gtagjs
+ *
+ * @api public
+ */
+
+GTAG.prototype.initialize = function() {
+  var config = [];
+  var that = this;
+  var gaOptions = this.options.gaOptions;
+  var GA_WEB_MEASUREMENT_ID = this.options.GA_WEB_MEASUREMENT_ID;
+  var GA_WEB_APP_MEASUREMENT_ID = this.options.GA_WEB_APP_MEASUREMENT_ID;
+  var AW_CONVERSION_ID = this.options.AW_CONVERSION_ID;
+  var DC_FLOODLIGHT_ID = this.options.DC_FLOODLIGHT_ID;
+  var accountId =
+    GA_WEB_MEASUREMENT_ID ||
+    GA_WEB_APP_MEASUREMENT_ID ||
+    AW_CONVERSION_ID ||
+    DC_FLOODLIGHT_ID;
+
+  if (GA_WEB_MEASUREMENT_ID) {
+    config.push(['config', GA_WEB_MEASUREMENT_ID]);
+    if (gaOptions && Object.keys(gaOptions).length) {
+      // set custom dimension and metrics if present
+      push('config', GA_WEB_MEASUREMENT_ID, {
+        custom_map: merge(gaOptions.dimensions, gaOptions.metrics)
+      });
+    }
+  }
+
+  if (GA_WEB_APP_MEASUREMENT_ID) {
+    config.push(['config', GA_WEB_APP_MEASUREMENT_ID]);
+    if (gaOptions && Object.keys(gaOptions).length) {
+      // set custom dimension and metrics if present
+      push('config', GA_WEB_APP_MEASUREMENT_ID, {
+        custom_map: merge(gaOptions.dimensions, gaOptions.metrics)
+      });
+    }
+  }
+
+  if (AW_CONVERSION_ID) {
+    config.push(['config', AW_CONVERSION_ID]);
+  }
+
+  if (DC_FLOODLIGHT_ID) {
+    config.push(['config', DC_FLOODLIGHT_ID]);
+  }
+
+  if (accountId) {
+    this.load({ accountId: accountId }, function() {
+      // Default routing.
+      for (var i = 0; i < config.length; i++) {
+        push(config[i][0], config[i][1]);
+      }
+      that.ready();
+    });
+  } else {
+    // Error case where not any of the ID specified
+  }
+};
+
+/**
+ * Loaded?
+ *
+ * @api private
+ * @return {boolean}
+ */
+
+GTAG.prototype.loaded = function() {
+  return !!(
+    window.gtagDataLayer && Array.prototype.push !== window.gtagDataLayer.push
+  );
+};
+
+/**
+ * Identify.
+ *
+ * @api public
+ * @param {Identify} event
+ */
+
+GTAG.prototype.identify = function(identify) {
+  var userId = identify.userId();
+  if (userId) {
+    if (this.options.GA_WEB_MEASUREMENT_ID) {
+      push('config', this.options.GA_WEB_MEASUREMENT_ID, {
+        user_id: userId
+      });
+    }
+    if (this.options.GA_WEB_APP_MEASUREMENT_ID) {
+      push('config', this.options.GA_WEB_APP_MEASUREMENT_ID, {
+        user_id: userId
+      });
+    }
+  }
```

Comment:
Same nit as above, can we return earlier to avoid nested code?
```suggestion
if (!userId) {
    return;
}

if (this.options.GA_WEB_MEASUREMENT_ID) {
    push('config', this.options.GA_WEB_MEASUREMENT_ID, {
      user_id: userId
    });
}

if (this.options.GA_WEB_APP_MEASUREMENT_ID) {
    push('config', this.options.GA_WEB_APP_MEASUREMENT_ID, {
        user_id: userId
    });
}
```

Reply:
Done